Src/Couchbase.Linq/QueryGeneration/DefaultMethodCallTranslatorProvider.cs

```diff
@@ -48,48 +52,61 @@ public IMethodCallTranslator GetTranslator(MethodCallExpression methodCallExpres
 
         protected virtual IMethodCallTranslator GetItem(MethodInfo key)
         {
-            if (key == null)
+            lock (Registry)
             {
-                throw new ArgumentNullException("key");
-            }
+                if (key == null)
+                {
+                    throw new ArgumentNullException("key");
+                }
 
-            IMethodCallTranslator translator;
-            if (Registry.TryGetValue(key, out translator))
-            {
-                return translator;
-            }
+                IMethodCallTranslator translator;
+                if (Registry.TryGetValue(key, out translator))
+                {
+                    return translator;
+                }
 
-            if (key.IsGenericMethod && !key.IsGenericMethodDefinition)
-            {
-                if (Registry.TryGetValue(key.GetGenericMethodDefinition(), out translator))
+                if (key.IsGenericMethod && !key.IsGenericMethodDefinition)
+                {
+                    if (Registry.TryGetValue(key.GetGenericMethodDefinition(), out translator))
+                    {
+                        return translator;
+                    }
+                }
+
+                // Check if the generic form of the declaring type matches
+                translator = GetItemFromGenericType(key);
+                if (translator != null)
                 {
                     return translator;
                 }
-            }
 
-            // Check if the generic form of the declaring type matches
-            translator = GetItemFromGenericType(key);
-            if (translator != null)
-            {
-                return translator;
-            }
+                // Check any interfaces that may have a matching method
+                translator = GetItemFromInterfaces(key);
+                if (translator != null)
+                {
+                    return translator;
+                }
 
```

Comment:
@jeffrymorris 

No extra \t, the entire method was indented to add the lock (Registry) wrapper.  I added this because we were modifying the registry, not just reading from it anymore.  Probably should have had it before, though.

Brant